lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java

```diff
@@ -297,23 +312,60 @@ void testQueryForExternalId() {
     return DynamicTest.stream(
         list.iterator(),
         t -> t.getLeft().name(),
-        t -> testQueryForCustomX(t.getLeft(), t.getMiddle(), t.getRight()));
+        t -> testQueryForCustomXLikeAndIn(t.getLeft(), t.getMiddle(), t.getRight()));
   }
 
-  void testQueryForCustomX(
+  void testQueryForCustomXLikeAndIn(
       TaskCustomField customField, String[] searchArguments, int expectedResult) throws Exception {
     List<TaskSummary> results =
         TASK_SERVICE.createTaskQuery().customAttributeLike(customField, searchArguments).list();
     assertThat(results).hasSize(expectedResult);
 
-    String[] ids =
+    String[] customAttributes =
         results.stream().map(t -> t.getCustomAttribute(customField)).toArray(String[]::new);
 
     List<TaskSummary> result2 =
-        TASK_SERVICE.createTaskQuery().customAttributeIn(customField, ids).list();
+        TASK_SERVICE.createTaskQuery().customAttributeIn(customField, customAttributes).list();
     assertThat(result2).hasSize(expectedResult);
   }
 
+  @WithAccessId(user = "admin")
+  @TestFactory
+  Stream<DynamicTest> should_ReturnCorrectResults_When_QueryingForCustomXNotIn() {
+    // carefully constructed to always return exactly 2 results
+    List<Pair<TaskCustomField, String[]>> list =
+        List.of(
+            Pair.of(TaskCustomField.CUSTOM_1, new String[] {"custom1"}),
+            Pair.of(TaskCustomField.CUSTOM_2, new String[] {"%"}),
+            Pair.of(TaskCustomField.CUSTOM_3, new String[] {"custom3"}),
+            Pair.of(TaskCustomField.CUSTOM_4, new String[] {"%"}),
+            Pair.of(TaskCustomField.CUSTOM_5, new String[] {"ew", "al", "el"}),
+            Pair.of(TaskCustomField.CUSTOM_6, new String[] {"11", "vvg"}),
+            Pair.of(TaskCustomField.CUSTOM_7, new String[] {"%"}),
+            Pair.of(TaskCustomField.CUSTOM_8, new String[] {"%"}),
+            Pair.of(TaskCustomField.CUSTOM_9, new String[] {"%"}),
+            Pair.of(TaskCustomField.CUSTOM_10, new String[] {"custom10"}),
+            Pair.of(TaskCustomField.CUSTOM_11, new String[] {"custom11"}),
+            Pair.of(TaskCustomField.CUSTOM_12, new String[] {"custom12"}),
+            Pair.of(TaskCustomField.CUSTOM_13, new String[] {"custom13"}),
+            Pair.of(TaskCustomField.CUSTOM_14, new String[] {"abc"}),
+            Pair.of(TaskCustomField.CUSTOM_15, new String[] {"custom15"}),
+            Pair.of(TaskCustomField.CUSTOM_16, new String[] {"custom16"}));
```

Comment:
Well done! 
I am aware that each result has only two hits. But I prefer a Triplet where the expected amount is viewable here.

Reason: If for some reason some test data gets added the number's will not match anymore